lib/utils.js

```diff
@@ -304,4 +310,32 @@ utils.mapHeaders = function mapHeaders(headers, addHeaders) {
     return headers;
 };
 
+utils.isUnsupportedQuery = function isUnsupportedQuery(queryStr) {
+    if (!queryStr) {
+        return false;
+    }
+    // Reject unsupported query fields (e.g. ?accelerate), but allow
+    // the same strings as values (e.g. ?test=accelerate) and queries like
+    // '?acceleratefoobar' or '&foobaraccelerate'
+    return constants.unsupportedQueries.some(q => {
```

Comment:
Rahul's comment was that it's more expensive to iterate through the keys of an object than to check if a string contains the values, and it's an actual concern since we can receive many requests with thousands of queries.